### ISSUE TITLE
win_chcolatey fails when successfull

### ISSUE THREAD

**Issue by lionbee** [ansible/ansible-modules-extras#990]
I'm using Ansible version 1.9.3

This command fails even though it is successful

``` yaml
- name: Install Microsoft Build Tools
  win_chocolatey:
    name: microsoft-build-tools -y
    upgrade: True
```

This is the error message:

> failed: [xxx.xxx.xxx.xxx] => {"changed": false, "chocolatey command": "choco.exe install microsoft-build-tools -y -source https://chocolatey.org/api/v2/", "failed": true}
>  msg: Install error: Chocolatey v0.9.9.8 Installing the following packages: microsoft-build-tools By installing you accept licenses for the packages.  microsoft-build-tools v12.0.21005.20140416  Downloading microsoft-build-tools 32 bit    from 'http://download.microsoft.com/download/9/B/B/9BB1309E-1A8F-4A47-A6C5-ECF76672A3B3/BuildTools_Full.exe'  Installing microsoft-build-tools...  microsoft-build-tools has been installed.  The install of microsoft-build-tools was successful.  Chocolatey installed 1/1 package(s). 0 package(s) failed.  See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

The problem stems from the fact that success is defined by string parsing instead of return code. Is there a reason return code is not used? I can understand that you will want to use string parsing to decide whether an install happened or whether the package was already present, but it seems you only want to raise an error if return code != 0

**Comment by robynbergeron** [comment]
[module: windows/win_chocolatey.py]

**Comment by robynbergeron** [comment]
Hi @thirty3 -- thanks for submitting this bug report.

@trondhindenes @petemounce @elventear @smadam813 -- pinging you to let you know about this bug report for win_chocolatey.

**Comment by tolbertam** [comment]
It appears that win_chocolatey has been refactored in 2.0 (#398) and it now checks status code as you suggest.  I dropped the version from the devel branch into stable1.9 locally and it looks it works completely as a drop in replacement.   I think it would be very nice if it got merged back to 1.9 otherwise this module will not work with newer versions of chocolatey.  Is there any reason why that couldn't be pulled in?  I would be a happy and willing tester! :+1:
